Triangulation_3/include/CGAL/Regular_triangulation_3.h

```diff
@@ -2388,40 +2453,96 @@ namespace CGAL {
 
   template < class Gt, class Tds, class Lds >
   bool
-    Regular_triangulation_3<Gt,Tds,Lds>::
-    remove(Vertex_handle v, bool *could_lock_zone)
+  Regular_triangulation_3<Gt,Tds,Lds>::
+  remove(Vertex_handle v, bool *could_lock_zone)
   {
     bool removed = true;
 
     // Locking vertex v...
-    if (!this->try_lock_vertex(v))
+    if(!this->try_lock_vertex(v))
     {
       *could_lock_zone = false;
     }
     else
     {
-      Vertex_handle hint = (v->cell()->vertex(0) == v ?
-        v->cell()->vertex(1) : v->cell()->vertex(0));
+      // Check that the vertex hasn't be deleted from the TDS while we were locking it
+#ifdef CGAL_RT3_IGNORE_TDS_CONCEPT
+      if(!tds().vertices().is_used(v))
+#else
+      if(!tds().is_vertex(v))
+#endif
```

Comment:
I am surprised that a vertex that we managed to lock can be destroyed by another thread. That would defeat all the purpose of the lock.

Reply:
I'm checking here that `v` is a valid vertex in case it was destroyed _before_ it was locked. I could also check the validity before the `try_lock_vertex`, but there's no real point since it might be destroyed in between that check and the lock.